File: healthcare_app_server/chat_service/models/call.js
const Room = require("./room");

class Call {
    constructor(id, bookAppointment, start, end, createdAt) {
        this._id = id;
        this._start = start;
        this._end = end;
        this._createdAt = createdAt;
        this._bookAppointment = bookAppointment;
    }


    get id() {
        return this._id;
    }

    set id(value) {
        this._id = value;
    }

    get bookAppointment() {
        return this._bookAppointment;
    }

    set bookAppointment(value) {
        this._bookAppointment = value;
    }

    get start() {
        return this._start;
    }

    set start(value) {
        this._start = value;
    }

    get end() {
        return this._end;
    }

    set end(value) {
        this._end = value;
    }

    get createdAt() {
        return this._createdAt;
    }

    set createdAt(value) {
        this._createdAt = value;
    }
}

module.exports = Call;

File: healthcare_app_server/admin_service/src/main/java/dev/skyherobrine/service/models/mongodb/Call.java
package dev.skyherobrine.service.models.mongodb;

import lombok.*;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;
import org.springframework.data.mongodb.core.mapping.FieldType;
import org.springframework.data.mongodb.core.mapping.MongoId;

import java.time.LocalDateTime;

@Document(collection = "calls")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
public class Call {
    @MongoId
    private Long id;
    private Appointment appointment;
    @Field(targetType = FieldType.DATE_TIME)
    private LocalDateTime start;
    @Field(targetType = FieldType.DATE_TIME)
    private LocalDateTime end;
    @Field(name = "created_at", targetType = FieldType.DATE_TIME)
    private LocalDateTime createdAt;
}
